jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -130,38 +130,43 @@ public interface Request extends Attributes, Content.Source
     String getId();
 
     /**
-     * @return the {@link Components} to be used with this request.
+     * @return the {@link Components} to be used with this {@code Request}.
      */
     Components getComponents();
 
     /**
-     * @return the {@code ConnectionMetaData} associated to this request
+     * @return the {@code ConnectionMetaData} associated to this {@code Request}
      */
     ConnectionMetaData getConnectionMetaData();
 
     /**
-     * @return the HTTP method of this request
+     * @return the HTTP method of this {@code Request}
      */
     String getMethod();
 
     /**
-     * @return the HTTP URI of this request
+     * @return the HTTP URI of this {@code Request}
      * @see #getContextPath(Request)
      * @see #getPathInContext(Request)
      */
     HttpURI getHttpURI();
 
     /**
+     * Get the {@link Context} associated with this {@code Request}.
+     * If the request is not being handled by a {@link org.eclipse.jetty.server.handler.ContextHandler} then
+     * the {@link org.eclipse.jetty.server.Server.ServerContext} instance will be returned.
      * @return the {@code Context} associated with this {@code Request}
      */
     Context getContext();
 
     /**
-     * <p>Returns the context path of this Request.</p>
-     * <p>This is equivalent to {@code request.getContext().getContextPath()}.</p>
+     * Get the context path of this {@code Request}.
+     * This is equivalent to {@code request.getContext().getContextPath()}.
      *
      * @param request The request to get the context path from.
-     * @return The contextPath of the request.
+     * @return The encoded context path of the {@link Context}.
+     *         If the request is not being handled by a {@link org.eclipse.jetty.server.handler.ContextHandler}
+     *         then {@code null} is returned (from {@link Server.ServerContext#getContextPath()}).
```

Comment:
Ditto.

Reply:
We need to explain that unlike getContext, this can be null.    I don't see the harm of being specific?    This issue was due to a users confusion with the null return.  I doubt they would have though oh this is a convenience call to another method that returns another class that has another method on which I should go read the javadoc.

Reply:
That is exactly what I experienced.

Mentioning ContextHandler is very helpful as it points me in the right direction if I want to "fix" this problem.

Reply:
See my other comment.

I am fine pointing to `ContextHandler`, but not go into the details of what subclass returns what here, where we want to be very abstract.

See for example the javadocs of `Request.getSession(boolean)`: it is generic, says it may return null but does not go into the details, and perhaps the only thing lacking is a pointer to `SessionHandler`, but the javadoc for the method is spot on.